8/src/part1.c

```diff
@@ -0,0 +1,102 @@
+// Link: https://adventofcode.com/2022/day/8
+
+#include <stdio.h>
+#include <stdbool.h>
+
+#define MAX_SIZE 500
+
+// Function for reading data from a file into a two-dimensional array.
+void read_tree_data(char filename[], int grid[][MAX_SIZE], int *rows, int *cols) {
+    FILE *file = fopen(filename, "r");
+    if (file == NULL) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    char line[MAX_SIZE];
+    *rows = 0;
+    while (fgets(line, sizeof(line), file) != NULL) {
+        for (*cols = 0; line[*cols] != '\n' && line[*cols] != '\0'; (*cols)++) {
+            grid[*rows][*cols] = line[*cols] - '0';// Change the character to an integer.
+        }
+        (*rows)++;
+    }
+
+    fclose(file);
+}
+
+bool is_tree_visible(int grid[][MAX_SIZE], int rows, int cols, int x, int y) {
+    int height = grid[x][y];
+
+    // сheck left
+    bool visible_from_left = true;
+    for (int i = y - 1; i >= 0; i--) {
+        if (grid[x][i] >= height) {
+            visible_from_left = false;
+            break;
+        }
+    }
+    if (visible_from_left) return true;
+
+    // сheck right
+    bool visible_from_right = true;
+    for (int i = y + 1; i < cols; i++) {
+        if (grid[x][i] >= height) {
+            visible_from_right = false;
+            break;
+        }
+    }
+    if (visible_from_right) return true;
```

Comment:
example
```suggestion
    int height = grid[x][y];

    // Check left
    for (int i = y - 1; i >= 0; i--) {
        if (grid[x][i] >= height) {
            break;
        }
        if (i == 0) return true;  // If we reach the edge without any taller trees, it's visible
    }

    // Check right
    for (int i = y + 1; i < cols; i++) {
        if (grid[x][i] >= height) {
            break;
        }
        if (i == cols - 1) return true;
    }
```

Reply:
I already cut off all the outermost trees from the function count_visible_trees. Like this: 
    for (int i = 1; i < rows - 1; i++) {
        for (int j = 1; j < cols - 1; j++) {
            if (is_tree_visible(grid, rows, cols, i, j)) {
                count++;
            }
        }
    }
 And just after finding the visibility of at least one side, the check stops. Putting them together can be hard since I'm only cycling from the current coordinate to the left (or right, up, down). going in both directions at once may require even more conditions and checks than there are now.